### PR TITLE
Update documentation on communication and testing

### PR DESCRIPTION
**Pull request by cboulanger**
This provides for more systematic and updated documentation on communication and testing. It is by no means comprehensive and more work is certainly needed, in particular in the testing subject area - we still need to revise the existing v5 docs here on how to write a unit test, on the sinon stuff, etc.

### CODE REVIEW

**Comment by cboulanger** [comment]
The following files have just been reformatted without editing the substance:
- remote_io.md
- request_io.md
- fake_server.md
- rest.md
- rpc.md (edited only the first few paragraphs)

Please comment only on the other files and provide your own PR on changes in these files after merging this PR.

**Review by level420 — APPROVED** [review]

File: docs/communication/fake_server.md [hunk 1]
```diff
@@ -1,79 +0,0 @@
-FakeServer
```
Comment (level420) [inline docs/communication/fake_server.md]: Why is the doc for the FakeServer deleted?

Comment (cboulanger) [reply docs/communication/fake_server.md]: It's moved to testing, where it belongs :-)

Comment (cboulanger) [reply docs/communication/fake_server.md]: https://github.com/qooxdoo/qooxdoo/pull/9890/files#diff-0ed61c13d9c4d30feeae53289d819841

**Review by oetiker — APPROVED** [review]

**Review by hkollmann — APPROVED** [review]

**Comment by cboulanger** [comment]
Quorum for documentation reached, merging.

**Pull request merged**